test/pthread/join_return_fail.c

```diff
@@ -0,0 +1,37 @@
+// Tests join & return in general - would fail if join doesn't block.
+//  Also tests that exit and join successfully store/load a return value
```

Comment:
I went through all of the files in the `test/pthread/` folder, and aligned all comments, and looked for non-conforming curly braces.

These will all appear in my next commit.